zebra/rt_netlink.c

```diff
@@ -1124,8 +1105,11 @@ static int build_label_stack(struct mpls_label_stack *nh_label,
  * @param nexthop: Nexthop information
  * @param nlmsg: nlmsghdr structure to fill in.
  * @param req_size: The size allocated for the message.
+ *
+ * The function returns true if the nexthop could be added
+ * to the message, otherwise false is returned.
  */
-static void _netlink_route_build_singlepath(const struct prefix *p,
+static bool _netlink_route_build_singlepath(const struct prefix *p,
```

Comment:
there's quite a bit of duplication between `build_singlepath()` and `build_multipath()` - do you think it might be possible to unify those at all?